appium/tests/specs/mock/setup/UpdateOlderPubKeyToNewer.spec.ts

```diff
@@ -0,0 +1,209 @@
+      await MenuBarScreen.clickMenuIcon();
+      await MenuBarScreen.checkUserEmail();
+
+      await MenuBarScreen.clickSettingsButton();
+      await SettingsScreen.checkSettingsScreen();
+      await SettingsScreen.clickOnSettingItem('Contacts');
+
+      await ContactScreen.checkContactScreen();
+      await ContactScreen.checkContact(userEmail);
+      // Go to Contact screen
+      await ContactScreen.clickOnContact(userEmail);
+      await ContactPublicKeyScreen.checkPgpUserId(userEmail);
+      await ContactPublicKeyScreen.checkPublicKeyDetailsNotEmpty();
+      await ContactPublicKeyScreen.clickOnFingerPrint();
+
+      await PublicKeyDetailsScreen.checkPublicKeyDetailsScreen();
+      await PublicKeyDetailsScreen.checkPublicKeyNotEmpty();
+      await PublicKeyDetailsScreen.checkSignatureDateValue(oldSignatureDate);
+      firstFetchedDate = await DataHelper.convertDateToMSec(await PublicKeyDetailsScreen.getLastFetchedDateValue());
+      await PublicKeyDetailsScreen.checkFingerPrintsValue(oldFingerprintsValue);
+
+      await PublicKeyDetailsScreen.clickBackButton();
+      await ContactPublicKeyScreen.checkPgpUserId(userEmail);
+      await ContactPublicKeyScreen.clickBackButton();
+
+      await ContactScreen.checkContactScreen();
+      await ContactScreen.clickBackButton();
+      await SettingsScreen.checkSettingsScreen();
+
+      await MenuBarScreen.clickMenuIcon();
+      await MenuBarScreen.clickInboxButton();
+      //stage 2
+      mockApi.attesterConfig = {
+        servedPubkeys: {
+          'updating.key@example.test': attesterPublicKeySamples.keyNewerVersion
+        }
+      };
+      await MailFolderScreen.checkInboxScreen();
+      await MailFolderScreen.clickCreateEmail();
+      await NewMessageScreen.setAddRecipient(userEmail);
+      await NewMessageScreen.checkAddedRecipient(userEmail);
+      await NewMessageScreen.clickBackButton();
+
+      // Go to Contacts screen
+      await MenuBarScreen.clickMenuIcon();
+      await MenuBarScreen.checkUserEmail();
+
+      await MenuBarScreen.clickSettingsButton();
+      await SettingsScreen.checkSettingsScreen();
+      await SettingsScreen.clickOnSettingItem('Contacts');
+
+      await ContactScreen.checkContactScreen();
+      await ContactScreen.checkContact(userEmail);
+      // Go to Contact screen
+      await ContactScreen.clickOnContact(userEmail);
+      await ContactPublicKeyScreen.checkPgpUserId(userEmail);
+      await ContactPublicKeyScreen.checkPublicKeyDetailsNotEmpty();
+      await ContactPublicKeyScreen.clickOnFingerPrint();
+
+      await PublicKeyDetailsScreen.checkPublicKeyDetailsScreen();
+      await PublicKeyDetailsScreen.checkPublicKeyNotEmpty();
+      await PublicKeyDetailsScreen.checkSignatureDateValue(newSignatureDate);
+      secondFetchedDate = await DataHelper.convertDateToMSec(await PublicKeyDetailsScreen.getLastFetchedDateValue());
+      await PublicKeyDetailsScreen.checkFingerPrintsValue(newFingerprintsValue);
```

Comment:
Is it possible to have helper functions to remove code duplication?

As currently checks on lines 47-73, 91-117, 137-163 are identical. It'll be great to have, for example, `checkSignatureAndFingerprints(userEmail, signatureDate, fingerprintsValue)` function which will perform all these checks. It'll make test more readable

Reply:
agree

Reply:
done